python/ray/tune/syncer.py

```diff
@@ -98,13 +97,12 @@ def validate_sync_config(sync_config: "SyncConfig"):
         sync_config.node_sync_period = -1
         sync_config.cloud_sync_period = -1
 
-        warnings.warn(
+        raise DeprecationWarning(
```

Comment:
These have been added to trigger a failure for demonstration purposes.

These changes will also be introduced in https://github.com/ray-project/ray/pull/24076 so after https://github.com/ray-project/ray/pull/24076 is merged these changes will disappear and we can just merge.